module/VuFind/src/VuFind/Recommend/Databases.php

```diff
@@ -230,6 +239,21 @@ public function getResults()
         }
         $nameToDatabase = $this->getDatabases();
         $databases = [];
+
+        // Add databases from search query
+        if ($this->useQuery) {
+            $query = strtolower($this->results->getParams()->getQuery()->getString());
```

Comment:
Should we perhaps have a minimum length requirement for this logic? Otherwise, I'm guessing that something like a search for a single letter is going to lead to very weird results.

Reply:
I was going to include that, but I considered that a lot of database names are acronyms, i.e. EPIC or JSTOR or are likely to be searched by partial names i.e. LEXIS or ABI (Inform), so in practice I can't see setting a minimum larger than like 3, and so what's the point.  But practicality inside I agree with the idea so can add it if you disagree.

Reply:
I think it might be worth making a configurable setting for it, with a default of 2 or 3, so people can decide for themselves. Acronyms aside, I also think blank searches may behave in an undesirable way without some kind of check (because str_contains always returns true if you ask for containment of the empty string -- but I would think the most commonly-used databases in the index would be a more useful default result than the first arbitrary databases listed in the configuration or API response).

Reply:
Good point about blank searches especially.  Implemented.